sql/moz-fx-data-shared-prod/telemetry_derived/cohort_daily_stats_clients_frst_seen_v2/query.sql

```diff
@@ -0,0 +1,167 @@
+-- Get all clients who had activity in the last 28 days and their bit patterns.
+WITH submission_date_activity AS (
+  SELECT
+    client_id,
+    days_seen_bits,
+    days_visited_1_uri_bits,
+    days_interacted_bits,
+    submission_date AS activity_date
+  FROM
+    telemetry.clients_last_seen_v1 -- this might cause an issue because definition of first_seen_date in this table is different from clients_first_seen_v2
+  WHERE
+    submission_date = DATE("2023-10-01")
+  GROUP BY
+    client_id,
+    submission_date,
+    days_seen_bits,
+    days_visited_1_uri_bits,
+    days_interacted_bits
+),
+-- Get all the cohorts that are still in range of the current day of activity (180 days)
+cohorts_in_range AS (
+  SELECT
+    client_id,
+    sample_id,
+    first_seen_date AS cohort_date,
+    second_seen_date,
+    -- DATE(@activity_date) AS activity_date,
+    DATE("2023-10-01") AS activity_date,
+    -- activity_segment, -- for desktop: from clients_last_seen, for mobile: calculated field from mobile_with_searches in unified_metrics
+    app_version,
+    attribution_campaign,
+    attribution_content,
+    attribution_experiment,
+    attribution_medium,
+    attribution_source,
+    attribution_ua,
+    -- attribution_variation, -- can find in clients_daily_v6 with unpacking attribution field
+    partner_distribution_version,
+    partner_distributor,
+    partner_distributor_channel,
+    partner_id,
+    city,
+    country,
+    -- device_model, -- important for mobile, usually null, may not need
+    apple_model_id, -- from clients_first_seen_v2
+    distribution_id,
+    -- is_default_browser,  -- can find in clients_daily_v6
+    locale,
+    -- normalized_app_name,  -- strings that come from unified_metrics/filters out BrowserStack
+    app_name as normalized_app_name, -- from clients_first_seen_v2, as per Lucia's msg app_name = normalized_app_name
+    normalized_channel,
+    normalized_os,
+    normalized_os_version,
+    -- os_version_major,
+    -- os_version_minor,
+    COALESCE(
+      SAFE_CAST(NULLIF(SPLIT(normalized_os_version, ".")[SAFE_OFFSET(0)], "") AS INTEGER),
+      0
+    ) AS os_version_major,
+    COALESCE(
+      SAFE_CAST(NULLIF(SPLIT(normalized_os_version, ".")[SAFE_OFFSET(1)], "") AS INTEGER),
+      0
+    ) AS os_version_minor,
+  FROM
+    -- telemetry_derived.rolling_cohorts_v1
+    telemetry_derived.clients_first_seen_v2
+  WHERE
+    -- cohort_date
+    first_seen_date
+    BETWEEN DATE_SUB(DATE("2023-10-01"), INTERVAL 180 DAY)
+    AND DATE_SUB(DATE("2023-10-01"), INTERVAL 1 DAY)
+    -- (1) No need to get activity for the cohort created on activity_date - everyone will be retained
```

Comment:
although, I don't think this is true anymore given how we are counting clients as retained based on the main ping. retention won't be 100% on the first day because some clients in CFS never send a main ping. so, there is a question whether we do actually want to calculate "retention" on first_seen_date = activity_date.

if we do, then people will be confused. if we don't, then we are technically dropping some information that MIGHT someday be useful.

Reply:
Are we using first_seen_date = activity_date elsewhere?